test/e2e/app-dir/unauthorized/basic/basic.test.ts

```diff
@@ -0,0 +1,41 @@
+import { nextTestSetup } from 'e2e-utils'
+
+describe('app dir - unauthorized - basic', () => {
+  const { next } = nextTestSetup({
+    files: __dirname,
+  })
+
+  it('should match dynamic route unauthorized boundary correctly', async () => {
+    // `/dynamic` display works
+    const browserDynamic = await next.browser('/dynamic')
+    expect(await browserDynamic.elementByCss('main').text()).toBe('dynamic')
+
+    // `/dynamic/401` calling unauthorized() will match the same level unauthorized boundary
+    const browserDynamic404 = await next.browser('/dynamic/401')
+    expect(await browserDynamic404.elementByCss('#unauthorized').text()).toBe(
```

Comment:
```suggestion
    const browserDynamic401 = await next.browser('/dynamic/401')
    expect(await browserDynamic401.elementByCss('#unauthorized').text()).toBe(
```

Reply:
appreciate the early review but it's still wip : )